rust/pact_models/src/path_exp.rs

```diff
@@ -35,39 +43,204 @@ fn matches_token(path_fragment: &str, path_token: &PathToken) -> usize {
   }
 }
 
-/// Calculates the path weight for a path expression and a given path. Returns a tuple of the
-/// calculated weight and the number of path tokens matched
-pub fn calc_path_weight(path_exp: &str, path: &[&str]) -> (usize, usize) {
-  let weight = match parse_path_exp(path_exp) {
-    Ok(path_tokens) => {
-      trace!("Calculating weight for path tokens '{:?}' and path '{:?}'", path_tokens, path);
-      if path.len() >= path_tokens.len() {
+#[derive(Debug, Clone, Eq, Serialize, Deserialize)]
+#[serde(try_from = "String")]
+#[serde(into = "String")]
+pub struct DocPath {
+  path_tokens: Vec<PathToken>,
+  expr: String,
+}
+
+impl DocPath {
+  pub fn new(expr: impl Into<String>) -> anyhow::Result<Self> {
+    let expr = expr.into();
+    let path_tokens = parse_path_exp(&expr)
+      .map_err(|e| anyhow!(e))?;
```

Comment:
If we update `parse_path_exp` to return an anyhow Result, we won't need to map anything here

Reply:
I see you have created a function for this: `first_field`

Reply:
Ok, happy to do that. I had initially avoided touching `parse_path_exp` to keep this already huge diff from endlessly expanding :)

Reply:
That's a fair comment. We can change that later.